g3doc/obtaining-training-data.md

```diff
@@ -0,0 +1,50 @@
+# Obtaining Training Data
+
+This document gives an overview of how to use the Gematria tooling to
+parse/process training data from multiple sources.
+
+## Parsing training data from the BHive dataset
+
+The publically available [BHive](https://github.com/ithemal/bhive) dataset
+has a decent amount of training data available in CSV format. Gematria
+provides native tooling to process these CSV files into `.tfrecord` files
+that can be used for further processing.
+
+For example, to process BHive throughput data for the Intel Skylake
+microarchitecture, we can begin by downloading one of the BHive CSVs:
+
+```bash
+curl -L https://raw.githubusercontent.com/ithemal/bhive/5f1d50077ac0779fd227b261dcf517862c7104bd/benchmark/throughput/skl.csv > skl.csv
+```
+
+If you look at the file, you'll notice that every line is formatted as
+`<machine code of BB in hex>,<throughput in cycles>>`. This is referred to as
+the BHive format. There are other projects composing similar datasets (such
+as Adreas Abel's [uica-eval](https://github.com/andreas-abel/uiCA-eval) which
+contains several improvements over the original BHive dataset) that use similar
+but not necessarily the same CSV format. Note that Gematria only supports the
+BHive format outlined above, so some preprocessing might be required depending
+upon the input data format.
+
+To process this into a `.tfrecord` that can be used, the `import_from_bhive`
+script within Gematria can be run through Bazel:
+
+```bash
+bazel run //gematria/datasets/python:import_from_bhive -- \
+    --gematria_input_csv=skl.csv \
+    --gematria_output_tfrecord=skl.tfrecord \
+    --gematria_throughput_source_name="bhive: skl"
+```
```

Comment:
I'd prefer to put both command-lines next to each other, so that the user can follow them like instructions, something like

1. Download the CSV file \
   ```bash
   curl -L https://raw.githubusercontent.com/ithemal/bhive/5f1d50077ac0779fd227b261dcf517862c7104bd/benchmark/throughput/skl.csv > skl.csv
   ```
3. Convert the CSV to `.tfrecord` \
   ```bash
   bazel run //gematria/datasets/python:import_from_bhive -- \
     --gematria_input_csv=skl.csv \
     --gematria_output_tfrecord=skl.tfrecord \
     --gematria_throughput_source_name="bhive: skl"
   ```